File: tests/fixtures/code/go/sample.go
// SPDX-FileCopyrightText: 2026 The Nikasha Authors
// SPDX-License-Identifier: Apache-2.0
package ring

import "fmt"

type Buffer struct {
	data []byte
	head int
}

type Sizer interface {
	Size() int
}

func New(n int) *Buffer {
	return &Buffer{data: make([]byte, n)}
}

func (b *Buffer) Push(x byte) {
	b.data[b.head] = x
	b.head = wrap(b.head+1, len(b.data))
}

func (b Buffer) Size() int {
	return len(b.data)
}

func wrap(i, n int) int {
	if i >= n {
		return 0
	}
	return i
}

func Dump(b *Buffer) {
	show := func() { fmt.Println(b.Size()) }
	show()
}

File: tests/fixtures/code/php/sample.php
<?php
// SPDX-FileCopyrightText: 2026 The Nikasha Authors
// SPDX-License-Identifier: Apache-2.0

namespace App\Util;

function slugify(string $s): string
{
    return strtolower(trim($s));
}

interface Renderer
{
    public function render(array $data): string;
}

class Page implements Renderer
{
    public function __construct(private string $title)
    {
    }

    public function render(array $data): string
    {
        $slug = slugify($this->title);
        return $this->wrap($slug) . Page::footer();
    }

    private function wrap(string $body): string
    {
        return \sprintf('<div>%s</div>', $body);
    }

    public static function footer(): string
    {
        return '';
    }
}

$page = new Page('Home');
echo $page?->render([]);
